mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
```

Comment:
The line `const result = await Order.findOneAndRemove({ _id: req.params.orderId });` has an issue. 

The `req.params.orderId` should be validated and sanitized to prevent NoSQL injection attacks. It's recommended to use a library like mongoose-validators or joi to validate the input. 

Additionally, it would be better to handle the case where `req.params.orderId` is not provided or is invalid. This could be done by adding a check before calling `Order.findOneAndRemove`.